### PR TITLE
Add configuration options for Firefox

### PR DESCRIPTION
**Pull request by Suvres**
these changes allow to edit preferences in Firefox

### CODE REVIEW

File: lib/Remote/DesiredCapabilities.php [hunk 1]
```diff
@@ -281,7 +281,7 @@ public static function chrome()
     /**
      * @return static
```
Comment (williamdes) [inline lib/Remote/DesiredCapabilities.php:282]:
```suggestion
     * @param array<string,string> $preferences Preferences to be added to the profile
     * @return static
```

File: tests/unit/Remote/DesiredCapabilitiesTest.php [hunk 2]
```diff
@@ -125,7 +125,8 @@ public function provideBrowserCapabilities()
 
     public function testShouldSetupFirefoxProfileAndDisableReaderViewForFirefoxBrowser()
     {
-        $capabilities = DesiredCapabilities::firefox();
+        $preferences = ['javascript.enabled' => false];
+        $capabilities = DesiredCapabilities::firefox($preferences);
```
Comment (williamdes) [inline tests/unit/Remote/DesiredCapabilitiesTest.php:129]: Adding an assertion that the capability was added would be a nice idea :)

**Comment by williamdes** [comment]
> Hi @williamdes ,
> When do You plan to merge it ?

Hi @Suvres 
I have no rights but added some reviewers that could merge your work

**Comment by Suvres** [comment]
Hi @andrefortin, @OndraM,
when You plan to review and merge it?

**Comment by OndraM** [comment]
Hi @Suvres, thanks for the pull request! I will have a look.

However you can already edit Firefox preferences like this, if you already need it with current version

```php
$capabilities = DesiredCapabilities::firefox();
$firefoxProfile = $capabilities->getCapability(FirefoxDriver::PROFILE);
$firefoxProfile->setPreference('foo.bar', false);
```

**Review by sanduluca — APPROVED** [review]

File: lib/Firefox/FirefoxProfile.php [hunk 3]
```diff
@@ -29,7 +29,7 @@ class FirefoxProfile
 
     /**
      * @param string $extension The path to the xpi extension.
-     * @return FirefoxProfile
+     * @return $this
```
Comment (williamdes) [inline lib/Firefox/FirefoxProfile.php:32]:
```suggestion
     * @return self
```
Would be better ?

**Comment by OndraM** [comment]
Hi,
I'm closing this pull request as redundant. The same could be easily done by setting `prefs` in `firefoxOptions`. I have added [wiki page for Firefox](https://github.com/php-webdriver/php-webdriver/wiki/Firefox#firefox-options), where this usage is covered with examples.

Thanks for your understanding

**Pull request closed without merging**